website/docs/gator.md

```diff
@@ -0,0 +1,183 @@
+
+- A Suite is a file which defines Tests.
+- A Test declares a ConstraintTemplate, a Constraint, and Cases to test the
+  Constraint.
+- A Case defines an object to validate and whether the object is expected to
+  pass validation.
+
+Any file paths declared in a Suite are assumed to be relative to the Suite
+itself. Absolute paths are not allowed. Thus, it is possible to move around a
+directory containing a Suite, and the files it uses for tests.
+
+### Suites
+
+[An example Suite file](https://github.com/open-policy-agent/gatekeeper-library/blob/8765ec11c12a523688ed77485c7a458df84266d6/library/general/allowedrepos/suite.yaml).
+
+To be valid, a Suite file must declare:
+```yaml
+kind: Suite
+apiVersion: test.gatekeeper.sh/v1alpha1
+```
+
+This differentiates Suites from "real" Kubernetes objects. `gator test` silently
+ignores files which do not declare these and does not attempt to run them.
+
+A Suite may declare multiple Tests, each containing different Templates and
+Constraints. Each Test in a Suite is independent.
+
+### Tests
+
+Each Suite contains a list of Tests under the `tests` field.
+
+A Test compiles a ConstraintTemplate, and instantiates a Constraint for the
+ConstraintTemplate. It is an error for the Constraint to have a different type
+than that defined in the ConstraintTemplate spec.crd.spec.names.kind, or for the
+ConstraintTemplate to not compile.
+
+### Cases
+
+Each Test contains a list of Cases under the `cases` field.
+
+A Case validates an object against a Constraint. The case may specify that the
+object is expected to pass or fail validation, and may make assertions about
+the returned violations (if any).
+
+A Case must specify `assertions` and whether it expects violations. The simplest
+way to declare this is:
+
+The Case expects at least one violation:
+```yaml
+assertions:
+- violations: yes
+```
+
+The Case expects no violations:
+```yaml
+assertions:
+- violations: no
+```
+
+- `violations` is either "yes", "no", or a non-negative integer. If "yes", at
+  least one violation must otherwise match the assertion. If "no", then no
+  violation messages must otherwise match the assertion. If a nonnegative
+  integer, then exactly that many violations must match.
+- `message` matches violations containing the exact string specified. `message`
+  is case-sensitive. If not specified or explicitly set to empty string, matches
+  all violations.
+
+A Case may specify multiple assertions. For example:
+
+```yaml
+  - name: both-disallowed
+    object: samples/repo-must-be-openpolicyagent/disallowed_both.yaml
+    assertions:
+    - violations: 2
+    - message: initContainer
+      violations: 1
+    - message: container
+      violations: 1
+```
+
+This Case specifies:
+
+- There are exactly two violations.
+- There is exactly one violation containing "initContainer".
+- There is exactly one violation containing "container".
+
+It is valid to check that no violations match an assertion. For example, the
```

Comment:
Could be better:

> It is valid to assert that no violations match a specified message.

This is more specific and may need to be edited, but I find it clearer.

Reply:
Done.